modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -167,6 +219,58 @@
 					this._processFile(imageFile, event.editor);
 				}
 			}
+			else if (event.data && event.data.type === 'html') {
+				const editor = event.editor;
+				const fragment = CKEDITOR.htmlParser.fragment.fromHtml(
+					event.data.dataValue
+				);
+
+				fragment.forEach((element) => {
+					if (
+						element.type === CKEDITOR.NODE_ELEMENT &&
+						element.name === 'img'
+					) {
+						const base64Regex = /^data:image\/(.*);base64,/;
+						const match = element.attributes.src.match(base64Regex);
+
+						if (element.attributes.src && match) {
+							const src = element.attributes.src;
+							const extension = match[1];
+							const name = `${Date.now().toString()}.${extension}`;
+
+							// Mark this "src" attribute as something we want to replace
+
+							this._imageSrcToReplace.push(src);
+
+							this._fetchBlob(src)
+								.then((blob) => {
+									const file = new File([blob], name, {
+										type: blob.type,
+									});
+
+									const el = CKEDITOR.dom.element.createFromHtml(
+										`<img src="${src}">`
+									);
+
+									const imageData = {
+										el,
+										file,
+									};
+
+									editor.fire('imageAdd', imageData);
```

Comment:
Another small nit: We could probably just pass it directly instead of creating this variable that is referenced exactly once:

```js
editor.fire('imageAdd', {
    el,
    file
});
```

Reply:
:+1: